book/chrome.md

```diff
@@ -800,51 +855,97 @@ class DrawOutline:
+            name = "Tab {}".format(i)
+            (tab_left, tab_top, tab_right, tab_bottom) = \
+                self.tab_bounds(i)
+
+            cmds.append(DrawLine(
+                tab_left, 0, tab_left, tab_bottom, "black", 1))
+            cmds.append(DrawLine(
+                tab_right, 0, tab_right, tab_bottom, "black", 1))
+            cmds.append(DrawText(
+                tab_left + self.padding, tab_top,
+                name, self.font, "black"))
+```
+
+Finally, to identify which tab is the active tab, we've got to make
+that file folder shape with the current tab sticking up:
+
+``` {.python}
+class Chrome:
+    def paint(self):
+        # ...
+        for i, tab in enumerate(self.browser.tabs):
+            # ...
+            if i == self.browser.active_tab:
+                cmds.append(DrawLine(
+                    0, tab_bottom, tab_left, tab_bottom, "black", 1))
+                cmds.append(DrawLine(
+                    tab_right, tab_bottom, WIDTH, tab_bottom,
+                    "black", 1))
+```
+
 The next step is clicking on tabs to switch between them. That has to
 happen in the `Browser` class, since it's the `Browser` that stores
 which tab is active. So let's go to the `handle_click` method and add
-a branch for clicking on the browser chrome:
+a branch for clicking on the browser chrome, which will delegate to a new
+method on the `Chrome` object.
 
 ``` {.python}
 class Browser:
     def handle_click(self, e):
-        if e.y < CHROME_PX:
-            # ...
+        if e.y < self.chrome.bottom:
+            self.chrome.click(e.x, e.y)
         else:
-            self.tabs[self.active_tab].click(e.x, e.y - CHROME_PX)
+            self.tabs[self.active_tab].click(
+                e.x, e.y - self.chrome.bottom)
         self.draw()
 ```
 
+``` {.python}
+class Chrome:
+    def click(self, x, y):
+        # ...
+```
+
 When the user clicks on the browser chrome (the `if` branch), the
 browser handles it directly, but if the click is on the page content
 (the `else` branch) it is still forwarded to the active tab,
-subtracting `CHROME_PX` to fix up the coordinates.
+subtracting `self.chrome.bottom` to fix up the coordinates.
 
-Within the browser chrome, the tab bar takes up the top 40 pixels,
-starting 40 pixels from the left. Remember that the `i`th tab has `x1
-= 40 + 80*i`; we need to solve that equation for `i` to figure out
-which tab the user clicked on:
+If the `y` coordinate is as high as `self.chrome.bottom`, it's just a matter
+of comparing it against each of the bounds methods we've already defined. To
+that end, let's add a quick method to test whether a point intersects one
+of them:
 
 ``` {.python}
-if e.y < CHROME_PX:
-    if 40 <= e.x < 40 + 80 * len(self.tabs) and 0 <= e.y < 40:
-        self.active_tab = int((e.x - 40) / 80)
+def intersects(x, y, rect):
+    (left, top, right, bottom) = rect
+    return x >= left and x < right and y >= top and y < bottom
```

Comment:
This is sort of neither here nor there but this method (and some general chaos in the `DrawXXX` methods) makes me think that we'd be better off defining a simple `Rect` class for the Tk chapters that transitions into `skia.Rect` once we move to Skia.